#! /bin/sh /usr/share/dpatch/dpatch-run
## 10_gtk-immodule-install-dir.dpatch by Ming Hua <minghua@rice.edu>
##
## All lines beginning with `## DP:' are a description of the patch.
##
## DP: Change the gtk immodule installation directory from
## DP: /usr/lib/gtk-2.0/immodules/ to
## DP: /usr/lib/gtk-2.0/$(GTK_BINARY_VERSION)/immodules/.
## DP:
## DP: The GTK+ IM module should be installed in the ABI-specific
## DP: directory instead of the general one to avoid ABI mismatch during
## DP: GTK+ ABI upgrade.
## DP:
## DP: Since GTK+ 2.10, Debian has changed the way to handle the ABI for
## DP: IM modules and GdkPixbuf loaders.  For IM modules, the
## DP: /usr/sbin/update-gtk-immodules tool is deprecated and the IM
## DP: module information is not stored in /etc/gtk-2.0/gtk.immodules
## DP: anymore.  The debhelper program dh_gtkmodules should be used
## DP: instead to update the IM module information.
## DP:
## DP: NOTE:  This patch only changes configure.ac and the change
## DP: necessary for configure is incorporated in
## DP: 90_relibtoolize.dpatch.  If the relibtoolize patch is removed,
## DP: this patch should change configure as well.

@DPATCH@
diff -urNad scim-1.4.7~/configure.ac scim-1.4.7/configure.ac
--- scim-1.4.7~/configure.ac	2007-06-26 09:31:57.000000000 -0500
+++ scim-1.4.7/configure.ac	2007-06-26 18:00:43.000000000 -0500
@@ -235,7 +235,7 @@
   if test -z "$GTK_LIBDIR)"; then
     GTK_LIBDIR="$libdir"
   fi
-  GTK_IM_MODULEDIR=$GTK_LIBDIR/gtk-2.0/immodules
+  GTK_IM_MODULEDIR=$GTK_LIBDIR/gtk-2.0/$GTK_BINARY_VERSION/immodules
   AC_SUBST(GTK_LIBDIR)
   AC_SUBST(GTK_IM_MODULEDIR)
   AC_SUBST(GTK_VERSION)
